plugins/generator-1.21.1/neoforge-1.21.1/templates/json/item.json.ftl

```diff
@@ -27,12 +27,20 @@
     <@modelDefinition/>
     </#if>
     <#macro modelDefinition>
-    "parent": "item/generated",
+	<#if data.hasCustomJAVAModel?? && data.hasCustomJAVAModel()>
+		<#assign parent = "builtin/entity">
+		<#assign texture = "particle">

```

Comment:
I don't fully understand the particle part. Do we really need to change this? Items don't have particles and also this works completely normally if I leave it at layer0.

Not having this would greatly simplify the code as you can inline the if